js/main.js

```diff
@@ -0,0 +1,119 @@
+(function() {
+
+    function loadPics() {
+        //supplied values from flickr
+        var api_key = '74e47a159e15cbcb6139ba9c9df64c13';
+        var secret = 'c73c5dcbac8e4a96';
+        var method = "flickr.photos.search";
+        var user_id = "138698049@N03";
+        var format = '&format=json';
+        var url = "https://api.flickr.com/services/rest"+"?method="+method+"&api_key="+api_key+"&user_id="+user_id+format+'&nojsoncallback=1';
+
+        var xmlhttp = new XMLHttpRequest();
+
+        xmlhttp.onreadystatechange = function () {
+            var farm, server, id, title, secret, picUrl, list, img, items;
+
```

Comment:
This is a little too busy for me as far as event handlers go.  I would recommend having a function that just handles taking an array of images and populating the`<ul>` goes.  This allows you to have a clear distinction between how you _get_ data, and how you _show_.  

Ideally, if it's supported, this where the concept of promises comes in handy.  Then there is never a need to have to call a render function as callback function in the data service function.
http://www.html5rocks.com/en/tutorials/es6/promises/